app/grandchallenge/reader_studies/views.py

```diff
@@ -1024,17 +1032,22 @@ def destroy(self, request, *args, **kwargs):
             )
         return super().destroy(request, *args, **kwargs)
 
-    def get_queryset(self):
-        queryset = (
-            DisplaySet.objects.all()
-            .select_related("reader_study__hanging_protocol")
-            .prefetch_related("values__image")
-        )
+    def filter_queryset(self, queryset):
+        # Note: if more fields besides 'reader_study' are added to the
+        # filter_set fields, we cannot call super anymore before randomizing
+        # as we only want to filter out the display sets for a specific
+        # reader study.
+        reader_study = self.reader_study
+        queryset = super().filter_queryset(queryset)
+        if reader_study and reader_study.shuffle_hanging_list:
+            set_seed(1 / int(self.request.user.pk))
+            queryset = queryset.order_by("?")
+            self.randomized_qs = list(queryset)
```

Comment:
Do you really need every value here? Might this be a huge request (and memory use)?

Reply:
We do need the full queryset, but I think we can get away with just storing the pks here now that I think of it.

Reply:
Can you add a test for the comment here?

Reply:
A test only to ensure that `reader_study` is the only filter_field?

Reply:
Yes, and referring the user back to this method if that does change.

Reply:
Using just the pks didn't work unfortunately, as this list is used later on if `unanswered_by_user=True`